src/test/java/pl/put/poznan/transformer/logic/serviceimpl/ScenarioCutToDepthServiceImplTest.java

```diff
@@ -0,0 +1,109 @@
+package pl.put.poznan.transformer.logic.serviceimpl;
+
+import org.junit.Before;
+import org.junit.Test;
+import pl.put.poznan.transformer.logic.domain.dto.ScenarioCutToDepthRequestDTO;
+import pl.put.poznan.transformer.logic.domain.dto.ScenarioDTO;
+import pl.put.poznan.transformer.logic.domain.dto.ScenarioStepDTO;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class ScenarioCutToDepthServiceImplTest {
+    private ScenarioCutToDepthServiceImpl scenarioCutToDepthService;
+
+    @Before
+    public void setUp() {
+        scenarioCutToDepthService = new ScenarioCutToDepthServiceImpl();
+    }
+
+    @Test
+    public void testCutScenarioToDepthLevelWorksForDepthLevelOne() {
+        ScenarioCutToDepthRequestDTO inputRequest = prepareInputCutToDepthRequest(1);
+        ScenarioDTO result = scenarioCutToDepthService.cutScenarioToDepthLevel(inputRequest);
+        assertScenarioSameAsInputScenarioCutToDepthLevelOne(inputRequest.getScenario(), result);
+    }
+
+    @Test
+    public void testCutScenarioToDepthLevelWorksForDepthLevelTwo() {
+        ScenarioCutToDepthRequestDTO inputRequest = prepareInputCutToDepthRequest(2);
+        ScenarioDTO result = scenarioCutToDepthService.cutScenarioToDepthLevel(inputRequest);
+        assertScenarioSameAsInputScenarioCutToDepthLevelTwo(inputRequest.getScenario(), result);
+    }
+
+    @Test
+    public void testCutScenarioToDepthLevelReturnsWholeScenarioWhenDepthLevelExceedsScenarioDepth() {
```

Comment:
Words of test name ought to be splitted by floor sign.

Reply:
In out project we have convention to use camel case. Look at other tests. I find underscores better only if they are used with given-when-then function naming convention to clearly separate these three phases.